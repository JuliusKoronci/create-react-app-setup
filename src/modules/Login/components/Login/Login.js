/* eslint-disable no-underscore-dangle */
// @flow
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class Login extends PureComponent {
  static defaultProps: Object;
  static propTypes: Object;

  render() {
    const { formValues, handleChange, handleSubmit } = this.props;
    return (
      <form>
        <input
          type="text"
          value={formValues._username}
          placeholder="Username"
          onChange={e => handleChange('_username', e.target.value)}
        />
        <input
          type="text"
          value={formValues._password}
          placeholder="Password"
          onChange={e => handleChange('_password', e.target.value)}
        />
        <button type="submit" onClick={handleSubmit}>Submit</button>
      </form>
    );
  }
}

Login.defaultProps = {
  formValues: {
    _username: '',
    _password: '',
  },
};
Login.propTypes = {
  formValues: PropTypes.shape({
    _username: PropTypes.string,
    _password: PropTypes.string,
  }).isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

export default Login;

