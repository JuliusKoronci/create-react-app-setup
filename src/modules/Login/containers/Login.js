// @flow
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import LoginView from '../components/Login';
import { actions, REDUCER_NAME } from '../ducks/loginDuck';
import formData from '../../../utils/formData';

export class Login extends Component {
  state = {
    formValues: {
      _username: '',
      _password: '',
    },
  };

  onChange = (key: string, value: string | number) => {
    this.setState({
      formValues: {
        ...this.state.formValues,
        [key]: value,
      },
    });
  };
  // eslint-disable-next-line no-undef
  onSubmit = (e: Event) => {
    e.preventDefault();
    const { formValues } = this.state;
    this.props.login({}, {}, formData(formValues));
  };

  render() {
    const { formValues } = this.state;
    return (
      <LoginView
        formValues={formValues}
        handleChange={this.onChange}
        handleSubmit={this.onSubmit}
      />
    );
  }
}

Login.propTypes = {
  login: PropTypes.func.isRequired,
};

function mapStoreToProps(state) {
  return {
    [REDUCER_NAME]: state[REDUCER_NAME],
  };
}
function mapDispatchToProps(dispatch) {
  return bindActionCreators({
    ...actions,
  }, dispatch);
}

export default connect(mapStoreToProps, mapDispatchToProps)(Login);
