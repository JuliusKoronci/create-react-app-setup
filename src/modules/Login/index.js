import loginReducer from './reducers/LoginReducer';
export { loginReducer };
