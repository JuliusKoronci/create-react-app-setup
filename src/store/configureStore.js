import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'react-router-redux';
import thunk from 'redux-thunk';
import createLogger from 'redux-logger';
import createHistory from 'history/createBrowserHistory';
import rootReducer from './rootReducer';
import DevTools from '../modules/DevTools';

export const history = createHistory();
const routerMiddlewareWithHistory = routerMiddleware(history);

function applyMiddlewareWithDevTool(...middleware) {
	if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) {
		return window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__(
			applyMiddleware(...middleware)
		);
	}

	return compose(
		applyMiddleware(...middleware),
		DevTools.instrument()
	);

}

const createDevStore = createStore(
	rootReducer,
	applyMiddlewareWithDevTool(routerMiddlewareWithHistory, thunk, createLogger)
);

const createProdStore = createStore(
	rootReducer,
	applyMiddleware(routerMiddlewareWithHistory, thunk)
);
const env = process.env.NODE_ENV;
const store = env === 'development' ? createDevStore : createProdStore;

if (module.hot) {
	module.hot.accept('./rootReducer', () => {
		const nextRootReducer = require('./rootReducer').default;
		store.replaceReducer(nextRootReducer);
	});
}

export default store;
