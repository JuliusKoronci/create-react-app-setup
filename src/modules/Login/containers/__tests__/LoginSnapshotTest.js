import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import Login from '../Login';

const middlewares = [];
const mockStore = configureStore(middlewares);

// Initialize mockstore with empty state
const initialState = {};
const store = mockStore(initialState);

const wrapper = shallow(
  <Login
    handleSubmit={() => {}}
    handleChange={() => {}}
    store={store}
  />,
);
describe('(Component) Login', () => {
  it('Should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

