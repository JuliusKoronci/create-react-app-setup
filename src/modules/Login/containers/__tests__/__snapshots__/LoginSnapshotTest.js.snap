// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Login Should match snapshot 1`] = `
<Login
  handleChange={[Function]}
  handleSubmit={[Function]}
  state={Object {}}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
